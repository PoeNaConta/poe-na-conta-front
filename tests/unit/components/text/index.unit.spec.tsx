import React from 'react';
import { render } from '@testing-library/react';
import Text, { TextProps } from '@components/text';

describe('Text', () => {
  it.each(['p', 'h1', 'h2', 'h3', 'h4', 'h5', 'h5', 'span'])(
    'should render element as %s',
    (element: TextProps['as']) => {
      const { container } = render(<Text as={element}>example text</Text>);

      expect(container).toMatchSnapshot();
    },
  );

  it.each(['xs', 's', 'm', 'l', 'xl', 'hero'])(
    'should render with size %s',
    (size: TextProps['size']) => {
      const { container } = render(<Text size={size}>example text</Text>);

      expect(container).toMatchSnapshot();
      expect(container.firstElementChild).toHaveClass('lds--text--' + size);
    },
  );

  it('should render with default props', () => {
    const { container } = render(<Text>example text</Text>);

    expect(container).toMatchSnapshot();
    expect(container.firstElementChild).toHaveClass('lds--text--md');
  });

  it.each([
    { primary: true },
    { secondary: true },
    { tertiary: true },
    { onBrand: true },
  ])('should render as %s', (props: TextProps) => {
    const { container } = render(<Text {...props}>example text</Text>);

    expect(container).toMatchSnapshot();
  });
});
