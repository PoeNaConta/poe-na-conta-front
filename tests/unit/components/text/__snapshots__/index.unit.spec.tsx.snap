// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Text > should render as { onBrand: true } 1`] = `
<div>
  <p
    class="lds--text lds--text--onbrand lds--text--md"
  >
    example text
  </p>
</div>
`;

exports[`Text > should render as { primary: true } 1`] = `
<div>
  <p
    class="lds--text lds--text--primary lds--text--md"
  >
    example text
  </p>
</div>
`;

exports[`Text > should render as { secondary: true } 1`] = `
<div>
  <p
    class="lds--text lds--text--secondary lds--text--md"
  >
    example text
  </p>
</div>
`;

exports[`Text > should render as { tertiary: true } 1`] = `
<div>
  <p
    class="lds--text lds--text--tertiary lds--text--md"
  >
    example text
  </p>
</div>
`;

exports[`Text > should render element as h1 1`] = `
<div>
  <h1
    class="lds--text lds--text--default lds--text--md"
  >
    example text
  </h1>
</div>
`;

exports[`Text > should render element as h2 1`] = `
<div>
  <h2
    class="lds--text lds--text--default lds--text--md"
  >
    example text
  </h2>
</div>
`;

exports[`Text > should render element as h3 1`] = `
<div>
  <h3
    class="lds--text lds--text--default lds--text--md"
  >
    example text
  </h3>
</div>
`;

exports[`Text > should render element as h4 1`] = `
<div>
  <h4
    class="lds--text lds--text--default lds--text--md"
  >
    example text
  </h4>
</div>
`;

exports[`Text > should render element as h5 1`] = `
<div>
  <h5
    class="lds--text lds--text--default lds--text--md"
  >
    example text
  </h5>
</div>
`;

exports[`Text > should render element as h5 2`] = `
<div>
  <h5
    class="lds--text lds--text--default lds--text--md"
  >
    example text
  </h5>
</div>
`;

exports[`Text > should render element as p 1`] = `
<div>
  <p
    class="lds--text lds--text--default lds--text--md"
  >
    example text
  </p>
</div>
`;

exports[`Text > should render element as span 1`] = `
<div>
  <span
    class="lds--text lds--text--default lds--text--md"
  >
    example text
  </span>
</div>
`;

exports[`Text > should render with default props 1`] = `
<div>
  <p
    class="lds--text lds--text--default lds--text--md"
  >
    example text
  </p>
</div>
`;

exports[`Text > should render with size hero 1`] = `
<div>
  <p
    class="lds--text lds--text--default lds--text--hero"
  >
    example text
  </p>
</div>
`;

exports[`Text > should render with size l 1`] = `
<div>
  <p
    class="lds--text lds--text--default lds--text--l"
  >
    example text
  </p>
</div>
`;

exports[`Text > should render with size m 1`] = `
<div>
  <p
    class="lds--text lds--text--default lds--text--m"
  >
    example text
  </p>
</div>
`;

exports[`Text > should render with size s 1`] = `
<div>
  <p
    class="lds--text lds--text--default lds--text--s"
  >
    example text
  </p>
</div>
`;

exports[`Text > should render with size xl 1`] = `
<div>
  <p
    class="lds--text lds--text--default lds--text--xl"
  >
    example text
  </p>
</div>
`;

exports[`Text > should render with size xs 1`] = `
<div>
  <p
    class="lds--text lds--text--default lds--text--xs"
  >
    example text
  </p>
</div>
`;
